src/RepoDriver.sol

```diff
@@ -0,0 +1,508 @@
+        /// @notice The requested repository owner updates.
+        mapping(bytes32 requestId => uint224 projectId) requestedUpdates;
+        /// @notice The new address of the AnyApi operator.
+        OperatorInterface operator;
+        /// @notice The fee in Link for each repository owner
+        /// update request when the driver is covering the cost.
+        uint96 defaultFee;
+        /// @notice The AnyApi job ID used for requesting repository owner updates.
+        bytes32 jobId;
+        /// @notice The AnyApi requests counter used as a nonce when calculating the request ID.
+        uint256 nonce;
+    }
+
+    /// @param _dripsHub The drips hub to use.
+    /// @param forwarder The ERC-2771 forwarder to trust. May be the zero address.
+    /// @param _driverId The driver ID to use when calling DripsHub.
+    constructor(DripsHub _dripsHub, address forwarder, uint32 _driverId)
+        ERC2771Context(forwarder)
+    {
+        dripsHub = _dripsHub;
+        driverId = _driverId;
+        // slither-disable-next-line uninitialized-local
+        address _linkToken;
+        if (block.chainid == CHAIN_ID_ETHEREUM) _linkToken = LINK_TOKEN_ETHEREUM;
+        else if (block.chainid == CHAIN_ID_GOERLI) _linkToken = LINK_TOKEN_GOERLI;
+        else revert("Unsupported chain");
+        linkToken = LinkTokenInterface(_linkToken);
+    }
+
+    modifier onlyRepoOwner(uint224 repoId) {
+        require(_msgSender() == repoOwner(repoId), "Caller is not the repo owner");
+        _;
+    }
+
+    /// @notice Calculates the user ID for the repository ID.
+    /// Every user ID is a 256-bit integer constructed by concatenating:
+    /// `driverId (32 bits) | repoId (224 bits)`.
+    /// @param repoId The ID of the repository.
+    /// @return userId The user ID.
+    function calcUserId(uint224 repoId) public view returns (uint256 userId) {
+        // By assignment we get `userId` value:
+        // `zeros (224 bits) | driverId (32 bits)`
+        userId = driverId;
+        // By bit shifting we get `userId` value:
+        // `driverId (32 bits) | zeros (224 bits)`
+        // By bit masking we get `userId` value:
+        // `driverId (32 bits) | repoId (224 bits)`
+        userId = (userId << 224) | uint224(repoId);
+    }
+
+    /// @notice Calculates the repository ID.
+    /// Every user ID is a 224-bit integer constructed by concatenating:
+    /// `forge (8 bits) | uint216(keccak256(name)) (216 bits)`.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @return repoId The ID of the repository.
+    function calcRepoId(Forge forge, bytes memory name) public pure returns (uint224 repoId) {
+        // By assignment we get `repoId` value:
+        // `zeros (216 bits) | forge (8 bits)`
+        repoId = uint8(forge);
+        // `nameHash` is the lower 27 bytes of the hash
+        uint216 nameHash = uint216(uint256(keccak256(name)));
```

Comment:
The Ethereum address is a 160-bit hash and if one can find a colliding preimage for a given address, it's much worse than here, 216 should be more than enough. This design allows having different repoId generation schemes for different forges. E.g. for GitHub it's a hash of the name, but for Heartwood it may be just the last 216 bits of the repo ID. On top of that you're right, it adds more information readily available in the repoId without making the design insecure or adding complexity, the only difference is whether you append the forge before or after hashing.

Reply:
There are some discussions around it: 
https://ethereum-magicians.org/t/increasing-address-size-from-20-to-32-bytes/5485/6

Reply:
If anybody finds a way to generate a preimage of a 160-bit keccak, they will generate a private key for `DripsHub`'s address and just steal all the tokens held by it anyway :shrug: The 216-bit repo identity will be a hollow fortress burned from the inside and abandoned. Making it a 224-bit or even a 256-bit won't save it.